corehq/apps/builds/utils.py

```diff
@@ -40,3 +43,71 @@ def extract_build_info_from_filename(content_disposition):
     else:
         raise ValueError('Could not find filename like {!r} in {!r}'.format(
             pattern, content_disposition))
+
+
+def get_latest_version_at_time(config, target_time, cache=None):
+    """
+    Get the latest CommCare version that was available at a given time.
+    Excludes superuser-only versions.
+    Menu items are already in chronological order (newest last).
+    If no target time is provided, return the latest version available now.
+
+    Args:
+        config: CommCareBuildConfig instance
+        target_time: datetime or string in ISO format, or None for latest version
+        cache: Dictionary for per-request caching
+    """
+    if cache is None:
+        cache = {}
+
+    if not target_time:
+        return config.get_default().version
+
+    if isinstance(target_time, str):
+        target_time = datetime.strptime(target_time, ISO_DATETIME_FORMAT)
```

Comment:
These lines are now unnecessary with the latest changes, right? We can be assured that `target_time` will be a `datetime` object?

Reply:
82bb7616018d97ba7eec526d941a0595f20cdf53